src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -306,10 +392,54 @@ public boolean equals(Object o)
      */
     public static void setRandomProfession(EntityVillager entity, Random rand)
     {
-        Set<String> entries = INSTANCE.professions.getKeys();
+        ArrayList<String> entries = new ArrayList(INSTANCE.professionsByName.keySet());
         int prof = rand.nextInt(entries.size());
-        //TODO: Grab id range from internal registry
-        entity.setProfession(rand.nextInt(5));
+        VillagerProfession profession = INSTANCE.professionsByName.get(entries.get(prof));
+        entity.setProfession(profession.id);
+    }
+    
+    public static void populateBuyingList(MerchantRecipeList recipeList, int professionID, int careerId, int careerLevel, Random random)
+    {
+        VillagerProfession profession = INSTANCE.professionsById.get(professionID);
+        //The villagers seem to have career = 0 meaning no career.
+        if(careerId > profession.careers.size())
+        {
+            return;
+        }
+        VillagerCareer career = profession.careers.get(careerId - 1);
+        career.populateBuyingList(recipeList, careerLevel, random);
+    }
+    
+    public static int getNumberOfCareers(int professionID)
+    {
+        return getProfessionForId(professionID).getCareerCount();
+    }
+    
+    public static VillagerProfession getProfessionForId(int professionID)
+    {
```

Comment:
those methods were mainly intended for use inside the EntityVillager, to keep the vanilla compatibility, should I instead make them take the EntityVillager itself?